Intersect.Client.Framework/Gwen/Input/InputHandler.cs

```diff
@@ -395,13 +396,44 @@ public static bool OnMouseClicked(Base canvas, int mouseButton, bool down)
         }
 
         /// <summary>
-        ///     Key handler.
+        ///     Mouse click handler.
         /// </summary>
         /// <param name="canvas">Canvas.</param>
-        /// <param name="key">Key.</param>
-        /// <param name="down">True if the key is down.</param>
+        /// <param name="mouseButton">Mouse button number.</param>
+        /// <param name="down">Specifies if the button is down.</param>
         /// <returns>True if handled.</returns>
-        public static bool OnKeyEvent(Base canvas, Key key, bool down)
+        public static bool OnMouseScroll(Base canvas, int deltaX, int deltaY)
+        {
+
+            if (canvas == null || 
+                HoveredControl.GetCanvas() != canvas ||
```

Comment:
Think we need to check to see if HoveredControl is null as they discovered here:
https://www.ascensiongamedev.com/topic/5267-kash-shop-mouse-scroll/?do=findComment&comment=47323

Otherwise I'm good with this.

Reply:
Ah, I only really looked at the original shop page. I'll get on that when I get out of bed.

Reply:
Updated this. 👍 
I wonder if it was related to the endless refresh loop of the friend list though, but it never hurts to add an additional check.